decimal.go

```diff
@@ -709,6 +709,29 @@ func (d Decimal) MarshalText() ([]byte, error) {
 	return []byte(d.String()), nil
 }
 
+// UnmarshalJSON implements the json.Unmarshaler interface.
+func (d *Decimal) UnmarshalJSON(s []byte) error {
+	var err error
+	unquoted := string(s[1 : len(s)-1])
+	*d, err = Parse(unquoted)
+	return err
+}
```

Comment:
Here you could call Unquote


```suggestion
// UnmarshalJSON implements the json.Unmarshaler interface.
func (d *Decimal) UnmarshalJSON(s []byte) error {
	var err error
	unquoted := strings.Unquote(string(s))
	*d, err = Parse(unquoted)
	return err
}
```

But also, you could simply do this


```suggestion
// UnmarshalText implements the [json.UnmarshalerText] interface.
func (d *Decimal) UnmarshalText(s []byte) error {
	var err error
	*d, err = Parse(string(s))
	return err
}
```